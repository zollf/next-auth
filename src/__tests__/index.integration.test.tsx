import React from 'react';
import { ReactWrapper, mount } from 'enzyme';
import client from 'next-auth/client';
import { Session } from 'next-auth';
import toJson from 'enzyme-to-json';

import Index from '@/pages';

const signOutMock = jest.spyOn(client, 'signOut');

const component = <Index />;

const mockSession: Session = {
  expires: '1',
  user: { email: 'a', name: 'Delta', image: 'c' },
};

describe('<Index />', () => {
  let wrapper!: ReactWrapper;
  beforeEach(() => {
    (client.useSession as jest.Mock).mockReturnValueOnce([undefined, false]);
    wrapper = mount(component);
  });

  it('matches its snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('is logged in', () => {
    expect(wrapper.find('Wrapper').props()['data-logged-in']).toBe(false);
  });

  describe('Logged in', () => {
    beforeEach(() => {
      (client.useSession as jest.Mock).mockReturnValueOnce([mockSession, false]);
      wrapper = mount(component);
    });

    it('matches its snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('is logged in', () => {
      expect(wrapper.find('Wrapper').props()['data-logged-in']).toBe(true);
    });

    it('sign out calls function', () => {
      wrapper.find('[data-test-id="logout"]').simulate('click');
      expect(signOutMock).toBeCalled();
    });
  });
});
