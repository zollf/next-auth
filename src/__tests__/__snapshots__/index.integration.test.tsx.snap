// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index /> Logged in matches its snapshot 1`] = `
<Index>
  <Wrapper
    data-logged-in={true}
  >
    <div
      className="wrapper"
    >
      <h1>
        Next Auth Login
      </h1>
      <p>
        Sign in as 
        Delta
      </p>
      <section>
        <Button
          href="/signIn"
        >
          <a
            className="anchor"
            href="/signIn"
          >
            <div
              className="button"
            >
              Sign In
            </div>
          </a>
        </Button>
        <Button
          href="/signUp"
        >
          <a
            className="anchor"
            href="/signUp"
          >
            <div
              className="button"
            >
              Sign Up
            </div>
          </a>
        </Button>
        <Button
          href="/api/users"
        >
          <a
            className="anchor"
            href="/api/users"
          >
            <div
              className="button"
            >
              View All Users
            </div>
          </a>
        </Button>
      </section>
      <Button
        data-test-id="logout"
        onClick={[Function]}
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          Logout
        </button>
      </Button>
    </div>
  </Wrapper>
</Index>
`;

exports[`<Index /> matches its snapshot 1`] = `
<Index>
  <Wrapper
    data-logged-in={false}
  >
    <div
      className="wrapper"
    >
      <h1>
        Next Auth Login
      </h1>
      <p>
        Not Signed In
      </p>
      <section>
        <Button
          href="/signIn"
        >
          <a
            className="anchor"
            href="/signIn"
          >
            <div
              className="button"
            >
              Sign In
            </div>
          </a>
        </Button>
        <Button
          href="/signUp"
        >
          <a
            className="anchor"
            href="/signUp"
          >
            <div
              className="button"
            >
              Sign Up
            </div>
          </a>
        </Button>
        <Button
          href="/api/users"
        >
          <a
            className="anchor"
            href="/api/users"
          >
            <div
              className="button"
            >
              View All Users
            </div>
          </a>
        </Button>
      </section>
    </div>
  </Wrapper>
</Index>
`;
